core/services/vrf/v1/integration_test.go

```diff
@@ -89,6 +89,10 @@ func TestIntegration_VRF_JPV2(t *testing.T) {
 			assert.NotNil(t, 0, runs[0].Outputs.Val)
 			assert.NotNil(t, 0, runs[1].Outputs.Val)
 
+			// stop jobs as to not cause a race condition in geth simulated backend
+			// between job creating new tx and fulfillment logs polling below
+			require.NoError(t, app.JobSpawner().DeleteJob(jb.ID))
```

Comment:
This will work if DeleteJob is guaranteed to be synchronous, i.e. that it does all its cleanup before returning.

Looking at the spawner, I think that is correct, although there's an open question here:
https://github.com/smartcontractkit/chainlink/blob/f53d37247ecfd0826e934b0ca109dd7d95c43543/core/services/job/spawner.go#L166 where we call service.Close(): there's no guarantee that all of those closes are synchronous, and you would have to check what's happening in the VRF case.

Reply:
This should be OK to rely on. They are meant to be synchronous. No guarantee, but anything that is not synchronous and is logging will eventually get flagged in a test as racey or logging after test completion.

Reply:
sgtm then